extensions/wikia/Helios/Exceptions.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Wikia\Helios;
+
+/**
+ * An exception class for the client.
+ */
+class Exception extends \Exception
+{
+	use \Wikia\Logger\Loggable;
+
+	private $data;
+
+	public function __construct( $message = null, $code = 0, \Exception $previous = null, $data = null ) {
+		parent::__construct( $message, $code, $previous );
+		$this->data = $data;
+		$this->logMe();
+	}
+
+	public function getData() {
+		return $this->data;
+	}
+
+	public function getResponse() {
+		return !empty($this->data) && !empty($this->data['response']) ? $this->data['response'] : null;
+	}
+
+	protected function logMe() {
+		// noop
+	}
```

Comment:
You can use a nice feature of `Loggable` trait - `getLoggerContext` method - and define the context that is logged by `logMe` methods in children classes.